Add documentation for CoreWCF extension on Upgrade Assistant (#30974)
dotnet/docs: docs/core/porting/upgrade-assistant-wcf.md

---
title: Upgrade WCF Server-side Project to use CoreWCF on .NET 6
description: Use the .NET Upgrade Assistant to upgrade an existing WCF Server-side project on .NET Framework to use CoreWCF services on .NET 6.
author: Simona Liao
ms.date: 09/01/2022
---
# Upgrade a WCF Server-side Project to use CoreWCF on .NET 6

The [.NET Upgrade Assistant](upgrade-assistant-overview.md) is a command-line tool that can assist with upgrading an existing WCF Server-side project on .NET Framework to use CoreWCF services on .NET 6. This article provides:

- Things to know before starting
- A demonstration of how to run the tool against a WCF Server-side project on .NET Framework
- Troubleshooting tips

For more information on how to install the tool, see [Overview of the .NET Upgrade Assistant](upgrade-assistant-overview.md).

## Things to know before starting

This tool currently supports C#, and uses [CoreWCF services](https://devblogs.microsoft.com/dotnet/corewcf-v1-released/) to port self-hosted WCF Server-side project.

Here are some requirements for the WCF project to be applicable for this upgrade:

1. Includes a .cs file that references `System.ServiceModel` and creates new `ServiceHost`.

>     a. If the WCF project has multiple `ServiceHost`, all hosts need to be created in the same method.

Reviewer: The "a." doesn't work as you think it does in markdown.

>     a. If the WCF project has multiple `ServiceHost`, all hosts need to be created in the same method.

Reviewer 2: already updated.

```
1. [Next step] Finalize upgrade

Choose a command:
   1. Apply next step (Finalize upgrade)
   2. Skip next step (Finalize upgrade)
   3. See more step details
   4. Configure logging
   5. Exit
>
[10:27:15 INF] Applying upgrade step Finalize upgrade
[10:27:15 INF] Upgrade step Finalize upgrade applied successfully
```

Ideally, after successfully running the tool, these modified files should look similar. Please check out the updated version of [CalculatorService.csproj](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/CalculatorService.csproj), [app.config](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/App.config), [wcf.config](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/wcf.config), [CalculatorSample/service.cs](https://github.com/dotnet/docs/tree/main/docs/core/porting/snippets/upgrade-assistant-wcf-framework/CalculatorSample.Upgraded/CalculatorService/service.cs).

> Notice that if you want to run both CalculatorClient and CalculatorService, you would need to upgrade CalculatorClient with the tool as well. Because CalculatorService targets .Net6.0 after upgrade, it cannot be referenced by CalculatorClient which still targets .NETFramework.

Reviewer: ```suggestion
Notice that if you want to run both CalculatorClient and CalculatorService, you would need to upgrade CalculatorClient with the tool as well. Because CalculatorService targets .Net6.0 after upgrade, it cannot be referenced by CalculatorClient which still targets .NET Framework.
```
What do you mean it can't be referenced by CalculatorClient? Is there a project to project reference or does the upgrade break compatibility with the existing client application.

Author: Actually, that's an outdated comment I will remove! Early on, the client project references the server project for the ICalculator ServiceContract. And then after upgrading the CalculatorService, the CalculatorClient has an error because it cannot reference CalculatorService since one is on .NET 6 one is on .NET Framework. But later I modified the client so there isn't a project to project reference and it won't have that error.

## After upgrading

> After you upgrade your projects, you'll need to compile and test them. Most certainly you'll have more work to do in finishing the upgrade. It's possible that the .NET Framework version of your app contained library references that your project isn't actually using. You'll need to analyze each reference and determine whether or not it's required. The tool may have also added or upgraded a NuGet package reference to wrong version.

Reviewer: ```suggestion
After you upgrade your projects, you'll need to compile and test them. The upgrade assistant will do what it can, but it can't solve every incompatibility as part of the project upgrade. For example, it's possible that the .NET Framework version of your app contained library references that your project isn't actually using. You'll need to analyze each reference and determine whether or not it's required. The tool may have also added or upgraded a NuGet package reference to wrong version.
```

## Troubleshooting tips

There are several known problems that can occur when using the .NET Upgrade Assistant. In some cases, these problems are with the [try-convert tool](https://github.com/dotnet/try-convert) that the .NET Upgrade Assistant uses internally.

[The tool's GitHub repository](https://github.com/dotnet/upgrade-assistant#troubleshooting-common-issues) has more troubleshooting tips and known issues.

## See also

> - [Upgrading a WCF service to .NET 6 with CoreWCF](https://devblogs.microsoft.com/dotnet/upgrading-a-wcf-service-to-dotnet-6/)

Reviewer: ```suggestion
- Blog: [Upgrading a WCF service to .NET 6 with CoreWCF](https://devblogs.microsoft.com/dotnet/upgrading-a-wcf-service-to-dotnet-6/)
```

> - [CoreWCF 1.0 has been Released, WCF for .NET Core and .NET 5+](https://devblogs.microsoft.com/dotnet/corewcf-v1-released/)

Reviewer: ```suggestion
- Blog: [CoreWCF 1.0 has been Released, WCF for .NET Core and .NET 5+](https://devblogs.microsoft.com/dotnet/corewcf-v1-released/)
```

> - [Overview of the .NET Upgrade Assistant](upgrade-assistant-overview.md)

Reviewer: ```suggestion
- Docs: [Overview of the .NET Upgrade Assistant](upgrade-assistant-overview.md)
```
